### PR TITLE
[Merged by Bors] - feat: port Archive.Imo.Imo1987Q1

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Archive/Imo/Imo1987Q1.lean [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+`∑ k : fin (card α + 1), k * p α k`. -/
+def fixedPointsEquiv' :
+    (Σ (k : Fin (card α + 1)) (σ : fiber α k), fixedPoints σ.1) ≃
+      { σx : α × Perm α // σx.2 σx.1 = σx.1 } where
+  toFun p := ⟨⟨p.2.2, p.2.1⟩, p.2.2.2⟩
+  invFun p :=
+    ⟨⟨card (fixedPoints p.1.2), (card_subtype_le _).trans_lt (Nat.lt_succ_self _)⟩, ⟨p.1.2, rfl⟩,
+      ⟨p.1.1, p.2⟩⟩
+  left_inv := fun ⟨⟨k, hk⟩, ⟨σ, hσ⟩, ⟨x, hx⟩⟩ =>
+    by
```
Comment (urkud) [inline Archive/Imo/Imo1987Q1.lean:92]:
```suggestion
  left_inv := fun ⟨⟨k, hk⟩, ⟨σ, hσ⟩, ⟨x, hx⟩⟩ => by
```

File: Archive/Imo/Imo1987Q1.lean [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+open scoped BigOperators Nat
+
+open Equiv Fintype Function
+
+open Finset (range sum_const)
+
+open Set (Iic)
+
+namespace Imo1987Q1
```
Comment (semorrison) [inline Archive/Imo/Imo1987Q1.lean:41]: Inconsistently namespaced: another file you did all caps. Let's pick one. :-) I suggest this one.

**Comment by semorrison** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**